client/client.go

```diff
@@ -22,57 +22,318 @@ limitations under the License.
 	for _, cont := range containerList {
-		cinfo := ContainerInfo{
-			Id:      cont.ID,
-			Status:  cont.Status,
-			Created: cont.Created,
-			Image:   cont.Image}
-		containers = append(containers, cinfo)
+		shortID, _ := GetShortId(cont.ID)
+		containers[shortID] = cont
+	}
+	containers["root"] = docker.APIContainers{ID: "/"}
+
+	if len(containers) == 0 {
+		return nil, errors.New("No docker container found")
 	}
+
 	return containers, nil
 }
+
+// isRunningSystemd returns true if the host was booted with systemd
+func isRunningSystemd() bool {
+	fi, err := os.Lstat("/run/systemd/system")
+	if err != nil {
+		return false
+	}
+	return fi.IsDir()
+}
+
+// isFsCgroupParent returns true if the docker was run with default cgroup parent
+func isFsCgroupParent(groupPath string) bool {
+	fi, err := os.Lstat(filepath.Join(groupPath, "docker"))
+	if err != nil {
+		return false
+	}
+	return fi.IsDir()
+}
+
+func isHost(id string) bool {
+	if id == "/" {
+		// it a host
+		return true
+	}
+
+	return false
+}
+
+func GetSubsystemPath(subsystem string, id string) (string, error) {
+	var subsystemPath string
+	systemSlice := "system.slice"
+
+	groupPath, err := cgroups.FindCgroupMountpoint(subsystem)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "[WARNING] Could not find mount point for %v\n", subsystem)
+		return "", err
+	}
+	if isHost(id) {
+		if isRunningSystemd() {
+			subsystemPath = filepath.Join(groupPath, systemSlice)
+		} else {
+			subsystemPath = groupPath
+		}
+		return subsystemPath, nil
+	}
+
+	if isFsCgroupParent(groupPath) {
+		// default cgroupfs parent is used for container
+		subsystemPath = filepath.Join(groupPath, "docker", id)
+	} else {
+		// cgroup is created under systemd.slice
+		subsystemPath = filepath.Join(groupPath, systemSlice, "docker-"+id+".scope")
+	}
+
+	return subsystemPath, nil
+}
+
+// GetStatsFromContainer returns docker containers stats: cgroups stats (cpu usage, memory usage, etc.) and network stats (tx_bytes, rx_bytes etc.)
+// Notes: incoming container id has to be full-length to be able to inspect container
+func (dc *dockerClient) GetStatsFromContainer(id string, collectFs bool) (*wrapper.Statistics, error) {
+	var (
+		stats      = wrapper.NewStatistics()
+		groupWrap  = wrapper.Cgroups2Stats // wrapper for cgroup name and interface for stats extraction
+		err        error
+		workingSet uint64
+	)
+	container := &docker.Container{}
+
+	var pid int
+
+	if !isHost(id) {
+		if !isFullLengthID(id) {
+			return stats, fmt.Errorf("Container id %+v is not fully-length - cannot inspect container", id)
+		}
+		// inspect container based only on fully-length container id.
+		container, err = dc.InspectContainer(id)
+
+		if err != nil {
+			// only log error message and return stats (contains cgroups stats)
+			return stats, nil
+		}
+		pid = container.State.Pid
+	}
+
+	for cg, stat := range groupWrap {
+		var groupPath string
+		var err error
+
+		groupPath, err = GetSubsystemPath(cg, id)
```

Comment:
Consider creating a new variables with a starting values:
``
groupPath, err := GetSubsystemPath(cg, id)

``
then [L247](https://github.com/intelsdi-x/snap-plugin-collector-docker/pull/32/files#diff-1cd41a2611da49ce63630d5a51bd82c5R247) and [L248](https://github.com/intelsdi-x/snap-plugin-collector-docker/pull/32/files#diff-1cd41a2611da49ce63630d5a51bd82c5R248) could be removed.